index.js

```diff
@@ -298,6 +304,77 @@ class GithubScm extends Scm {
         }
     }
 
+    /**
+     * Generate a deploy private and public key pair
+     * @async  generateDeployKey
+     * @return {Promise}                    Resolves to object containing the public and private key pair
+     */
+    async generateDeployKey() {
+        return new Promise((resolve, reject) => {
+            const location = DEPLOY_KEYS_FILE;
+            const comment = this.config.email;
+            const password = DEPLOY_KEYS_PASSWORD;
+            const format = DEPLOY_KEYS_FORMAT;
+
+            keygen({
+                location,
+                comment,
+                password,
+                read: true,
+                format
+            }, (err, keyPair) => {
+                if (err) {
+                    logger.error('Failed to create keys: ', err);
+
+                    return reject(err);
+                }
+
+                return resolve(keyPair);
+            });
+        });
+    }
+
+    /**
+     * Adds deploy public key to the github repo and returns the private key
+     * @async  _addDeployKey
+     * @param  {Object}     config
+     * @param  {Object}     config.token        Admin token for repo
+     * @param  {String}     config.checkoutUrl  The checkoutUrl to parse
+     * @return {Promise}                        Resolves to the private key string
+     */
+    async _addDeployKey(config) {
+        const { token, checkoutUrl } = config;
+        const scmInfo = getInfo(checkoutUrl);
+        const keys = await this.generateDeployKey();
+
+        try {
+            await this.breaker.runCommand({
+                action: 'createDeployKey',
+                scopeType: 'repos',
+                token,
+                params: { owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    title: DEPLOY_KEY_TITLE,
+                    key: keys.pubKey,
+                    read_only: true }
+            });
```

Comment:
suggest using object destructuring to simplify the code a little; also some indentation tweaks
```suggestion
        const { owner, repo } = getInfo(checkoutUrl);
        const { pubKey } = await this.generateDeployKey();

        try {
            await this.breaker.runCommand({
                action: 'createDeployKey',
                scopeType: 'repos',
                token,
                params: {
                    owner,
                    repo,
                    title: DEPLOY_KEY_TITLE,
                    key: pubKey,
                    read_only: true
                }
            });
```